opentelemetry-sdk/tests/test_configurator.py

```diff
@@ -665,9 +677,151 @@ def test_logging_init_exporter(self):
         getLogger(__name__).error("hello")
         self.assertTrue(provider.processor.exporter.export_called)
 
+    @patch.dict(environ, {}, clear=True)
+    def test_otel_log_level_by_name_default(self):
+        self.assertEqual(_get_log_level(), NOTSET)
+
+    @patch.dict(environ, {"OTEL_LOG_LEVEL": "NOTSET "}, clear=True)
+    def test_otel_log_level_by_name_notset(self):
+        self.assertEqual(_get_log_level(), NOTSET)
+
+    @patch.dict(environ, {"OTEL_LOG_LEVEL": " DeBug "}, clear=True)
+    def test_otel_log_level_by_name_debug(self):
+        self.assertEqual(_get_log_level(), DEBUG)
+
+    @patch.dict(environ, {"OTEL_LOG_LEVEL": " info "}, clear=True)
+    def test_otel_log_level_by_name_info(self):
+        self.assertEqual(_get_log_level(), INFO)
+
+    @patch.dict(environ, {"OTEL_LOG_LEVEL": " warn"}, clear=True)
+    def test_otel_log_level_by_name_warn(self):
+        self.assertEqual(_get_log_level(), WARNING)
+
+    @patch.dict(environ, {"OTEL_LOG_LEVEL": " warnING "}, clear=True)
+    def test_otel_log_level_by_name_warning(self):
+        self.assertEqual(_get_log_level(), WARNING)
+
+    @patch.dict(environ, {"OTEL_LOG_LEVEL": " eRroR"}, clear=True)
+    def test_otel_log_level_by_name_error(self):
+        self.assertEqual(_get_log_level(), ERROR)
+
+
     @patch.dict(
         environ,
-        {"OTEL_RESOURCE_ATTRIBUTES": "service.name=otlp-service"},
+        {
+            "OTEL_RESOURCE_ATTRIBUTES": "service.name=otlp-service",
+            "OTEL_LOG_LEVEL": "CUSTOM_LOG_LEVEL",
+        },
+        clear=True,
+    )
+    @patch("opentelemetry.sdk._configuration._get_log_level", return_value=39)
+    def test_logging_init_exporter_level_under(self, log_level_mock):
+        # log_level_mock.return_value = 39
+        resource = Resource.create({})
+        _init_logging(
+            {"otlp": DummyOTLPLogExporter},
+            resource=resource,
+        )
+        self.assertEqual(self.set_provider_mock.call_count, 1)
+        provider = self.set_provider_mock.call_args[0][0]
+        self.assertIsInstance(provider, DummyLoggerProvider)
```

Comment:
What do you think about these assertions? Do we want to keep just the ones that test the changes in this PR?

Reply:
I see this as testing a use case of _init_logging so I kept those to ensure that the new features do not affect anything else. But I'm good either way.

Reply:
I'll reorder the tests to make that clearer.